python/mxnet/model.py

```diff
@@ -789,6 +819,117 @@ def fit(self, X, y=None, eval_data=None, eval_metric='acc',
                             sym_gen=self.sym_gen)
 
 
+    def profiling(self, X, y=None, eval_data=None, eval_metric='acc',
```

Comment:
I think
```
mx.profiler.turn_on(...)
model.fit()
```
should work. Why do you want to start profiling in the middle?

Also, since we are deprecating FeedForward and moving to Module, consider implementing this in module. Module supports calling fit for multiple times each time with intervels like fit(0, 10); profiler.on(); fit(10, 11)